wgsl/index.bs

```diff
@@ -3477,6 +3492,10 @@ loop_statement
 </pre>
 
 The loop construct causes a block of statements, the *loop body*, to execute repeatedly.
+A declaration introduced in a loop statement is [=in scope=] until the end of
```

Comment:
Scope is a textual (static) concept, but "end of the current loop statemen'ts _current_ iteration" is a dynamic concept.

Suggest: 

> The loop statement causes a [=compound statement=], the `<dfn noexport>loop body</dfn>` to execute repeatedly.
> Each execution of the loop body is called an `<dfn noexport>iteration</dfn>`.

> The identifier for a declaration in a loop is [=in scope=] from immediately after the declaration statement until the end of the loop body. 
> The declaration is executed each time it is reached, so each new iteration creates a new instance of the variable or constant, and re-evaluates the initializer expression.

Reply:
The reason loops are treated specially (besides iterations) is that the grammar doesn't define the body as a block/compound statement. We could change the grammar to make all block/compound statements the same and add rules that prevent the use of break, continue, fallthrough and continuing generally, or we could say they are special cases of block/compound statements.

Reply:
Ah, I see.  I got that for case clauses, but not for loop.  I was only looking at the diff, and not the grammar for loop (shown in the full spec).

How about refactoring so we make a 'loop_body' concept in the grammar:

       loop_statement
         :  LOOP  BRACE_LEFT loop_body BRACE_RIGHT

       loop_body
         : statements continuing_statement?

>The loop statement causes a set of statements, the `<dfn noexport>loop body</dfn>`, to execute repeatedly.
> Each execution of the loop body is called an `<dfn noexport>iteration</dfn>`.

> The identifier for a declaration in a loop body is [=in scope=] from immediately after the declaration statement until the end of the loop body. 
> The declaration is executed each time it is reached, so each new iteration creates a new instance of the variable or constant, and re-evaluates its initializer expression, if it has one.

Reply:
ah. I see. Sorry.